### PR TITLE
[Modular] Some clean up for Modular tests

### PR DESCRIPTION
**Pull request by DN6**
# What does this PR do?
Some changes to make modular tests more in line with our new testing approach

1. Remove `unittest` dependency 
2. Move `get_pipeline` to ModularPipelineTesterMixin (it's most likely going to be the same across all pipelines)
3. Create distinction between tests and test mixins. Since we don't use `unittest` anymore, we need to be explicit about which objects are Mixins and which objects are actually running the tests and need attributes such as `param`, `repo` etc. 
4. Always use `cpu` based Generator through the `self.get_generator` method in `ModularPipelineTesterMixin` for better reproducibility (GPU based generator can introduce noise)
5. Use `torch.abs().max` for comparing slices. Let's move away from `np` conversions. There isn't much of an advantage and when we run on GPU we can avoid `.detach().cpu().numpy()` calls 
6. Use single level inheritance for easier debugging. 
 

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: tests/modular_pipelines/test_modular_pipelines_common.py [hunk 1]
```diff
@@ -57,9 +47,8 @@ class ModularPipelineTesterMixin:
         ]
     )
 
-    def get_generator(self, seed):
-        device = torch_device if torch_device != "mps" else "cpu"
-        generator = torch.Generator(device).manual_seed(seed)
+    def get_generator(self, seed=0):
+        generator = torch.Generator("cpu").manual_seed(seed)
```
Comment (sayakpaul) [inline tests/modular_pipelines/test_modular_pipelines_common.py:51]: (nit): `torch.manual_seed(seed)` is also enough.

**Comment by sayakpaul** [comment]
I will propagate this to #12585! Thanks for this PR
